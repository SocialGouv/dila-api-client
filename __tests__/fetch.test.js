const fetch = require("jest-fetch-mock");

jest.setMock("node-fetch", fetch);

jest.mock("simple-oauth2", () => ({
  create: () => ({
    clientCredentials: {
      getToken: () => {}
    },
    accessToken: {
      create: () => ({
        token: { access_token: 1234 }
      })
    }
  })
}));

fetch.mockResponse(JSON.stringify({ message: "YATTA!" }));

const Client = require("..");

describe("client.fetch", () => {
  test("make a call with correct token and parameters", async () => {
    const client = new Client();
    const res = await client.fetch({
      path: "consult/code/tableMatieres",
      method: "POST",
      params: {
        date: 1234,
        sctId: "",
        textId: "LEGITEXT000006072050"
      }
    });
    expect(fetch.mock.calls.length).toEqual(1);
    const firstCall = fetch.mock.calls[0];
    expect(firstCall[0]).toEqual(
      `${client.apiHost}/consult/code/tableMatieres`
    );
    expect(firstCall[1]).toMatchSnapshot();
  });
});
