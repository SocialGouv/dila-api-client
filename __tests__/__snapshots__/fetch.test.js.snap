// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`client.fetch make a call with correct token and parameters 1`] = `
Object {
  "body": "{\\"date\\":1234,\\"sctId\\":\\"\\",\\"textId\\":\\"LEGITEXT000006072050\\"}",
  "headers": Object {
    "Authorization": "Bearer 1234",
    "content-type": "application/json",
  },
  "method": "POST",
}
`;
